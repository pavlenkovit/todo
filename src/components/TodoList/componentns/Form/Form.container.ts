import { connect } from 'react-redux';

import Form from './Form';
import { addTask } from '../../../../store/actions';

export interface IDispatchFromPropsForm {
  addTask: (value: string) => void;
}

const mapDispatchToProps = (dispatch: any) => {
  return {
    addTask: (value: string) => dispatch(addTask({ value })),
  };
};

export default connect<null, IDispatchFromPropsForm, void>(
  null, mapDispatchToProps
)(Form);
