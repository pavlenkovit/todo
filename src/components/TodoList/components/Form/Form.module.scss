.container {
  margin: 15px 0;
  display: flex;
}

.input {
  margin-right: 15px;
  flex: 1 1 auto;
  width: 1%;
  border: solid 2px #000;
  font-size: 20px;
  border-radius: 3px;
  padding: 0.35em 0.5em;
}

.button {
  width: 120px;
  border-radius: 3px;
  background: #414141;
  color: #fff;
  font-size: 20px;
  font-weight: 600;
  cursor: pointer;
}
