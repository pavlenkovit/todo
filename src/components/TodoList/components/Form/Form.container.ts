import { connect } from 'react-redux';

import Form from './Form';
import { addTask } from '../../../../store/actions/tasks';

export interface IDispatchFromPropsForm {
  addTask: (value: string) => void;
}

const mapDispatchToProps = { addTask };

export default connect<null, IDispatchFromPropsForm, void>(
  null, mapDispatchToProps,
)(Form);
