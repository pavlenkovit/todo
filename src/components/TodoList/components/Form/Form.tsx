import React, { useState } from 'react';
import styles from './Form.module.scss';
import { IDispatchFromPropsForm } from './Form.container';

const Form: React.SFC<IDispatchFromPropsForm> = ({ addTask }) => {
  const [value, onChange] = useState('');

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    onChange(e.target.value);
  };

  const onSubmit = () => {
    if (value === '') {
      return;
    }
    addTask(value);
    onChange('');
  };

  return (
    <div className={styles.container}>
      <input type="text" onChange={handleChange} value={value} className={styles.input} />
      <button onClick={onSubmit} className={styles.button}>Add</button>
    </div>
  );
};

export default Form;
