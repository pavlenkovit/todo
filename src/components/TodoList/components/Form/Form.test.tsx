import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Form from './Form';
import { IDispatchFromPropsForm } from './Form.container';

configure({ adapter: new Adapter() });

describe('<Form />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((value) => [value, onChange]);

  const props: IDispatchFromPropsForm = {
    addTask: () => {},
  };

  const form = shallow(<Form {...props} />);

  it('renders properly', () => {
    expect(form).toMatchSnapshot();
  });

  describe('Count Up', () => {
    it('calls setCount with count + 1', () => {
      form.find('#button').props().onClick();
      expect(setState).toHaveBeenCalledWith('');
    });
  });

});
