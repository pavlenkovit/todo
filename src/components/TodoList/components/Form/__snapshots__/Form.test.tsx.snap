// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders properly 1`] = `ShallowWrapper {}`;

exports[`Form renders properly 1`] = `ShallowWrapper {}`;
